src/query_ctx.c

```diff
@@ -135,6 +150,7 @@ static void _QueryCtx_ThreadSafeContextUnlock(QueryCtx *ctx) {
 
 bool QueryCtx_LockForCommit(void) {
```

Comment:
Please remind me, what will happen if this function gets called from the redis main thread, is it ok to try and lock the GIL ?

Reply:
it checks if it is executed from a blocking client, and if so, it locks the GIL, otherwise it is called from Redis main thread and will not lock.